R2021b blog page (#3212)
cyberbotics/webots: docs/blog/Webots-2021-b-release.md

An example showing the integration of OpenAI Gym with Webots is now available in our samples library.

---

## Reduce Package Size

For this release we work hard at finding solutions to reduce the size of the Webots installation packages.
To this end, we removed textures, meshes and sounds from the package. These resources are now downloaded from the web only when needed and cached for subsequent use.
Additionally, we removed the offline documentation and ..

TODO (update with package size difference).

---

## New Web Rendering Engine

Since the introduction of the web interface for streaming simulations and playing recorded animations, we struggled finding optimal parameters for shadows and illuminations that will reproduce exactly the same rendering as in the Webots Desktop application.
Finally, we decided to port our custom 3D renderer WREN to WebAssembly and use it for the web simulation interface in order to minimize the differences with the Desktop version and make it straightforward to run high quality simulations and animations on the web.

> During this work, we also took the time to refresh and improve the interface.

Reviewer: ```suggestion
During this work, we also took the time to refresh and improve the graphical user interface.
```

> %figure "New Wen Simulation Interface"

Reviewer: ```suggestion
%figure "New Web Simulation Interface"
```

![Web Simulation Interface](images/web_simulation_interface.png)
%end

---

## Extended Motor and Joint Functionalities

The [Motor](../reference/motor.md) and [Joint](../reference/joint.md) nodes have been extended to provide more realistic simulations.

> We enhanced the `Motor` node to support coupled motors specifying the linked motor in the `name` field and setting their ratio using the new `multiplier` field.

Reviewer: ```suggestion
We enhanced the [Motor](../reference/motor.md) node to support coupled motors specifying the linked motor in the `name` field and setting their ratio using the new `multiplier` field.
```

> Then, we added two new PROTO nodes [HingeJointWithBacklash](../guide/hinge-joint-with-backlash) and [Hinge2JointWithBacklash](../guide/hinge-2-joint-with-backlash) to easily add backlash effect and a [Gear](..//doc/guide/object-gear.md) model to simulate collision-based transmission.

Reviewer: ```suggestion
Then, we added two new PROTO nodes: [HingeJointWithBacklash](../guide/hinge-joint-with-backlash) and [Hinge2JointWithBacklash](../guide/hinge-2-joint-with-backlash) to easily add backlash effect and a [Gear](..//doc/guide/object-gear.md) model to simulate collision-based transmission.
```

Author: I would not add the semi colon here. Otherwise the sentence doesn't make sense anymore given that it also specifies the gear proto.

Reviewer: OK, then this is probably better:
```suggestion
Then, we added three new PROTO nodes. The [HingeJointWithBacklash](../guide/hinge-joint-with-backlash) and [Hinge2JointWithBacklash](../guide/hinge-2-joint-with-backlash) allow users to easily add a backlash effect in hinges. The [Gear](..//doc/guide/object-gear.md) model can be used to simulate a collision-based transmission.
```

Author: I would rather split the two sentences.
I would change it when I'm back.

We've added many new robot models in this release.

### Modular Mobile Platforms

We just added three new models of modular mobile platform robots used for logistic and indoor transport.
All these models have predefined fields to mount additional devices on the platform and front/back lidars.

TODO show the three robot figures on a single line

#### [Robotnik Summit-XL Steel](../guide/summit-xl-steel.md)
%figure "Summit-XL Steel in Webots"
![Summit-XL Steel in Webots](images/summit_xl_steel.wbt.thumbnail.jpg)
%end

#### [Mobile Industrial Robots MiR100](../guide/mir100.md)
%figure "MiR100 in Webots"
![MiR100 in Webots](images/mir100.wbt.thumbnail.jpg)
%end

#### [REC Fabtino](../guide/fabtino.md)

%figure "Fabtino in Webots"
![Fabtino in Webots](images/fabtino.wbt.thumbnail.jpg)
%end

### NVIDIA JetBot

The NVIDIA [JetBot](../guide/jetbot.md) is an affordable two-wheeled robot based on the NVIDIA Jetson Nano board widely used in AI robotics applications.

%figure "JetBot in Webots"
![JetBot in Webots](images/jetbot.wbt.thumbnail.jpg)
%end

A standalone model of the NVIDIA [Jetson Nano](../guide/single-board-computers.md#nvidia-jetson-nano) board is also available.

> 

Reviewer: We should also add the Niryo Ned robot here.

---

## ROS 2 & Webots

TODO

---

## Extra Goodies

> A new `Billboard` node has been introduced to display information in the main view regardless of the positoin and orientation of the viewpoint.

Reviewer: We should like Billboard to the documentation page here and maybe also the Viewpoint.

Reviewer 2: ```suggestion
A new `Billboard` node has been introduced to display information in the main view regardless of the position and orientation of the viewpoint.
```

> The [SICK S300](../guide/lidar-sensors.md#sick-s300) lidar model is now included in our sensors library.

Reviewer: A picture of the SICK S300 would be nice here.

Author: Usually we never added pictures in the extra goodies section. 
If we want to add a picture, then it is better to create a new section for new sensors.

Reviewer: Then, I believe it deserves a new sensors section.

> We added JavaScript scripting support for procedural PROTO nodes.

Reviewer: A link to the documentation page would be nice here.
```suggestion
We added JavaScript scripting support for procedural PROTO nodes in addition to LUA.
```

| OS | R2021a Package Size | R2021b Package Size |
| :---: | :---: | :---: |
| Windows | 1.65 GB | 448 MB |
| Linux | 1.45 GB | 263 MB |
| macOS | 1.42 GB | 226 MB |

---

## New Web Rendering Engine

Since the introduction of the web interface for streaming simulations and playing recorded animations, we struggled finding optimal parameters for shadows and illuminations that will reproduce exactly the same rendering as in the Webots Desktop application.
Finally, we decided to port our custom 3D renderer WREN to WebAssembly and use it for the web simulation interface in order to minimize the differences with the Desktop version and make it straightforward to run high quality simulations and animations on the web.

During this work, we also took the time to refresh and improve the graphical user interface.

%figure "New Web Simulation Interface"
![Web Simulation Interface](images/web_simulation_interface.png)
%end

---

## Extended Motor and Joint Functionalities

The [Motor](../reference/motor.md) and [Joint](../reference/joint.md) nodes have been extended to provide more realistic simulations.

> We enhanced the [Motor](../reference/motor.md) node to support coupled motors specifying the linked motor in the `name` field and setting their ratio using the new `multiplier` field.

Reviewer 3: ```suggestion
We enhanced the [Motor](../reference/motor.md) node to support coupled motors specifying the linked motor in the `name` field and setting their ratio using the new `multiplier` field.
This can simplify the implementation of multi-fingered grippers.
```

Reviewer 3: I've put it as a suggestion, but feel free to dismiss it if you don't think it's necessary.

Then, we added three new PROTO nodes. The [HingeJointWithBacklash](../guide/hinge-joint-with-backlash.md) and [Hinge2JointWithBacklash](../guide/hinge-2-joint-with-backlash.md) allow users to easily add a backlash effect in hinges. The [Gear](../guide/object-gear.md) model can be used to simulate a collision-based transmission.

%figure "Coupled Motors with Gears"

> ![Coupled Motors With Gears](images/coupled_motors.gif)

Reviewer 3: Just to be clear, in this gif the gears are purely visual, they're not used as a transmission. This demo is about the coupled motors where only one of the two fingers of the gripper is commanded directly and the other receives it by relay.

It's clear for me, but if it's what was assumed then it might be worth rephrasing it.

Author: Yes, it was also clear for me.
I think that the `Gear` PROTO is also useful just for graphical effect and not only for physical interaction.

But I will specify it so that there is no misunderstanding.

| | | |
| :---: | :---: | :---: |
| ![Summit-XL Steel](images/summit_xl_steel.wbt.thumbnail.jpg) | ![MiR100](images/mir100.wbt.thumbnail.jpg) | ![Fabtino](images/fabtino.wbt.thumbnail.jpg) |
| [Robotnik Summit-XL Steel](../guide/summit-xl-steel.md) | [Mobile Industrial Robots MiR100](../guide/mir100.md) | [REC Fabtino](../guide/fabtino.md) |

### Nyrio Ned

The Nyrio Ned is a 6-axis collaborative robot designed for teaching programming in an industrial environment.

![Nyrio Ned in Webots video](https://www.youtube.com/watch?v=diBAJY1WJPQ)

### NVIDIA JetBot

The NVIDIA [JetBot](../guide/jetbot.md) is an affordable two-wheeled robot based on the NVIDIA Jetson Nano board widely used in AI robotics applications.

%figure "JetBot in Webots"
![JetBot in Webots](images/jetbot.wbt.png)
%end

A standalone model of the NVIDIA [Jetson Nano](../guide/single-board-computers.md#nvidia-jetson-nano) board is also available.

---

## New Sensors

The [SICK S300](../guide/lidar-sensors.md#sick-s300) lidar model is now included in our sensors library.

%figure "SICK S300"
![SICK S300](images/sick_s300.thumbnail.png)
%end

> Additionally, Webots R2021b now provides a new [Altimeter](../reference/altimeter.md) sensor to determine the global altitude of a robot or robot part.

Reviewer 3: ```suggestion
Additionally, thanks to community contributor [sishamilton](https://github.com/sishamilton), Webots R2021b now provides a new [Altimeter](../reference/altimeter.md) sensor to determine the global altitude of a robot or robot part.
```

Reviewer 3: Attribution

We haven't forgotten about the old good ROS 1.
It now integrates the [ros\_control](http://wiki.ros.org/ros_control) framework and you don't have to activate sensors with services anymore.
These improvements make Webots much simpler to integrate into ROS 1 applications.
We also included a Tiago example that features SLAM and MoveIt integration.

%figure "Webots integration with ROS, ros_control, gmapping, and MoveIt."
![Webots ROS Tiago](images/webots_ros_tiago.gif)
%end

---

## Extra Goodies

A new [Billboard](../reference/billboard.md) node has been introduced to display information in the main view regardless of the position and orientation of the viewpoint.

We added [JavaScript scripting support](../reference/javascript-procedural-proto.md) for procedural PROTO nodes in addition to LUA.

**Go and [download](https://cyberbotics.com/#download) Webots R2021b today, so you don't miss out on all these great new features!**

---

## Acknowledgements

> The current release includes contributions from [Adman](https://github.com/Adman), [Aethor](https://github.com/Aethor), [AleBurzio11](https://github.com/AleBurzio11), [aykborstelmann](https://github.com/aykborstelmann), [angel-ayala](https://github.com/angel-ayala), [ar0usel](https://github.com/ar0usel), [bkpcoding](https://github.com/bkpcoding), [Behrouz-m](https://github.com/Behrouz-m), [BruceXSK](https://github.com/BruceXSK), [correll](https://github.com/correll), [DavidMansolino](https://github.com/DavidMansolino), [doggo4242](https://github.com/doggo4242), [Dorteel](https://github.com/Dorteel), [fmrico](https://github.com/fmrico), [it14149](https://github.com/it14149), [jmarsik](https://github.com/jmarsik), [kyax32](https://github.com/kyax32), [medrimonia](https://github.com/medrimonia), [NicolasGuy97](https://github.com/NicolasGuy97), [renan028](https://github.com/renan028), [ShuffleWire](https://github.com/ShuffleWire), [sigmunau](https://github.com/sigmunau), [Simon-Steinmann](https://github.com/Simon-Steinmann) and [WasabiFan](https://github.com/WasabiFan).

Reviewer 3: ```suggestion
The current release includes contributions from [Adman](https://github.com/Adman), [Aethor](https://github.com/Aethor), [AleBurzio11](https://github.com/AleBurzio11), [aykborstelmann](https://github.com/aykborstelmann), [angel-ayala](https://github.com/angel-ayala), [ar0usel](https://github.com/ar0usel), [bkpcoding](https://github.com/bkpcoding), [Behrouz-m](https://github.com/Behrouz-m), [BruceXSK](https://github.com/BruceXSK), [correll](https://github.com/correll), [DavidMansolino](https://github.com/DavidMansolino), [doggo4242](https://github.com/doggo4242), [Dorteel](https://github.com/Dorteel), [fmrico](https://github.com/fmrico), [it14149](https://github.com/it14149), [jmarsik](https://github.com/jmarsik), [kyax32](https://github.com/kyax32), [medrimonia](https://github.com/medrimonia), [NicolasGuy97](https://github.com/NicolasGuy97), [renan028](https://github.com/renan028), [ShuffleWire](https://github.com/ShuffleWire), [sigmunau](https://github.com/sigmunau), [Simon-Steinmann](https://github.com/Simon-Steinmann), [sishamilton](https://github.com/sishamilton) and [WasabiFan](https://github.com/WasabiFan).
```

Reviewer 3: Add contributor here as well.
